Update guide_mailman-3.rst - works (almost) without qmail

frafl opened the pull request:
Add an alternative configuration which does not rely on qmail (except for two minor options).

nichtmax commented:
Hey @frafl, thank you for your PR! We'll take a look at it and test it the next days.

jskvbinmv commented on source/guide_mailman-3.rst:
```diff
@@ -464,6 +464,100 @@ Install cronjobs
+    #END of server options
+    #
+    #BEGIN user options
+    username isabell-catchalldefault
```
This should be isabell-chatchallbox for consistency.

frafl replied:
Sorry, yes. I used `-catchalldefault` for my own account, but thought `catchallbox` would be a better general fit.

jskvbinmv commented on source/guide_mailman-3.rst:
```diff
@@ -464,6 +464,100 @@ Install cronjobs
+    #for whatever reason 'ssl' is a user option and must come *after* username
+    ssl
+    idle
+    password ""
```
password "CATCHALLPWD"

frafl replied:
Correct.

jskvbinmv commented on source/guide_mailman-3.rst:
```diff
@@ -464,6 +464,100 @@ Install cronjobs
+Finally, we also need an supervisord unit file for fetchmail, which we will place at ``/home/isabell/etc/services.d/fetchmail.ini``:
+
+.. code:: ini
+
```
Missing header: [program:fetchmail]

frafl replied:
Correct, I'll change that.

jskvbinmv commented on source/guide_mailman-3.rst:
```diff
@@ -464,6 +464,100 @@ Install cronjobs
+
+.. code-block:: bash
+
+  servicectl reread
```
Shouldn't this be supervisorctl?

frafl replied:
Yes, sorry, `supervisorctl` is of course correct.

jskvbinmv replied:
Also I just noticed that there is a

`supervisorctl update
`

missing

frafl replied:
Oh, that's even more important. This explains, why I had to `reload`. I checked my history, `update` was indeed missing.

frafl replied:
OK

jskvbinmv reviewed (commented):
I've just tested this and found some inconsistencies in the guide.

jskvbinmv commented:
I tried this approach but couldn't get it to work.



```
/home/isabell/var/logs/fetchmail_err.log

The nodetach option is in effect, ignoring logfile option.
fetchmail: connection to isabell.local.uberspace.de:8024 [fd75:6272:7370:12e::2/8024] failed: Connection refused.
fetchmail: LMTP error: 550 Requested action not taken: mailbox unavailable
fetchmail: LMTP listener doesn't like recipient address `isabell@isabell.local.uberspace.de'
fetchmail: can't even send to isabell!
```

frafl commented:
@jskvbinmv 
> fetchmail: connection to isabell.local.uberspace.de:8024 [fd75:6272:7370:12e::2/8024] failed: Connection refused.

Could you also check the `_out.log`? My log shows failed attempts with the IPv6 address, but a successful connection using IPv4 directly after that.

frafl commented:
@jskvbinmv 
> fetchmail: LMTP error: 550 Requested action not taken: mailbox unavailable
> fetchmail: LMTP listener doesn't like recipient address `isabell@isabell.local.uberspace.de'
> fetchmail: can't even send to isabell!

This is not directly related to an initial error, but a a try to send a status mail to "your" account isabell after another error occurred. Fetchmail assumes that this can be done via the same LMTP connection, but this fails of course.

The  `_out.log` file would be helpful here, too. A successfully handled mail looks like this:

```
fetchmail: 1 is first unseen
1 message for isabell-catchallbox at stardust.uberspace.de (folder Mailman).
fetchmail: IMAP> A1762 FETCH 1 RFC822.SIZE
fetchmail: IMAP< * 1 FETCH (RFC822.SIZE 365)
fetchmail: IMAP< A1762 OK Fetch completed (0.001 + 0.000 secs).
fetchmail: IMAP> A1763 FETCH 1 RFC822.HEADER
fetchmail: IMAP< * 1 FETCH (RFC822.HEADER {365}
reading message isabell-catchallbox@stardust.uberspace.de:1 of 1 (365 header octets) About to rewrite Return-Path: <isabell@stardust.uberspace.de>...
...rewritten version is Return-Path: <isabell@stardust.uberspace.de>.
About to rewrite From: isabell@stardust.uberspace.de...
...rewritten version is From: isabell@stardust.uberspace.de.
About to rewrite Cc: recipient list not shown: ;...
...rewritten version is Cc: recipient list not shown: ;.
fetchmail: Parsing envelope "Delivered-To" names "Delivered-To: isabell-mymailinglist@isabell.uber.space"
fetchmail: passed through mymailinglist@isabell.uber.space matching isabell.uber.space

Trying to connect to fd75:6272:7370:12e::2/8024...connection failed.
Trying to connect to [IPv4 Adress]  ...connected.
fetchmail: LMTP< 220 stardust.uberspace.de GNU Mailman LMTP runner 2.0
fetchmail: LMTP> LHLO stardust.uberspace.de
fetchmail: LMTP< 250-stardust.uberspace.de
fetchmail: LMTP< 250-SIZE 33554432
fetchmail: LMTP< 250-8BITMIME
fetchmail: LMTP< 250 HELP
fetchmail: forwarding to isabell.local.uberspace.de/8024
fetchmail: LMTP> MAIL FROM:<isabell@stardust.uberspace.de> SIZE=365
fetchmail: LMTP< 250 OK
fetchmail: LMTP> RCPT TO:<mymailinglist@isabell.uber.space>
fetchmail: LMTP< 250 Ok
fetchmail: LMTP> DATA
fetchmail: LMTP< 354 End data with <CR><LF>.<CR><LF>
```

This assumes that mailing list has address `mymailinglist@isabell.uber.space`. At least the domain of the mailing list address must end in `isabell.uber.space` (`subdomain.isabell.uber.space` is also fine), otherwise the configuration option ` localdomains isabell.uber.space` must be adapted to contain (a suffix of) the domain. This options is crucial for multi drop mode. Without it no username replacement is done.

Maybe you could also add a `postmaster` address or the option `no bouncemail` (I'll leave an inline comment), to stop  fetchmail from sending to isabell via LMTP. I did both of this after finishing the initial version of this guide.

frafl commented on source/guide_mailman-3.rst:
```diff
@@ -464,6 +464,100 @@ Install cronjobs
+  set no syslog
+  #has no effect with --nodetach, but ensures that we log to the same file, when we debug manually
+  set logfile /home/isabell/var/logs/fetchmail_out.log
+
```
2 further options:
```
set no bouncemail
set postmaster isabell@uber.space
```
This sends error messages to another address and does not use the LMTP connection for that, because this is a global option and LMTP is server+user specific.

jskvbinmv replied:
> This assumes that mailing list has address mymailinglist@isabell.uber.space. At least the domain of the mailing list address must end in isabell.uber.space (subdomain.isabell.uber.space is also fine), otherwise the configuration option  localdomains isabell.uber.space must be adapted to contain (a suffix of) the domain. 

That was the problem I left it at isabell.uber.space. Changing it to lists.mydomain.tld resolved the error.

Thanks for your work on this! As a guide follower I couldn't have figured out on my own how to get rid of qmail files.

This also fixes https://github.com/Uberspace/lab/issues/1741

jskvbinmv commented:
> This assumes that mailing list has address mymailinglist@isabell.uber.space. At least the domain of the mailing list address must end in isabell.uber.space (subdomain.isabell.uber.space is also fine), otherwise the configuration option  localdomains isabell.uber.space must be adapted to contain (a suffix of) the domain. 

That was the problem I left it at isabell.uber.space. Changing it to lists.mydomain.tld resolved the error.

Thanks for your work on this! As I am just a guide follower I couldn't have figured out on my own how to get rid of qmail files.

This also fixes https://github.com/Uberspace/lab/issues/1741

kimdiallo commented:
Please excuse the long time you had to wait for the merge. We have written out qmail and revised the structure a little.

nichtmax commented:
@frafl please get in touch via hallo@uberspace.de for your well earned goodie!
